### PR TITLE
feat: add inference config api

### PR DESCRIPTION
**Pull request by zhuangqh**
**Reason for Change**:

- API change: add config to `workspace.inference.config`
- generate a default config if no user config specified by copying from release-namespace

**Requirements**

- [x] added unit tests and e2e tests (if applicable).

### CODE REVIEW

File: pkg/workspace/inference/preset-inferences.go [hunk 1]
```diff
@@ -191,3 +203,70 @@ func CreatePresetInference(ctx context.Context, workspaceObj *kaitov1alpha1.Work
 	}
 	return depObj, nil
 }
+
```
Comment (Fei-Guo) [inline pkg/workspace/inference/preset-inferences.go:206]: Is there any chance to dedup this method code with the existing code in fine tuning for the similar funcitonality?

**Review by Fei-Guo — COMMENTED** [review]
where is the default inference configmap?

**Comment by zhuangqh** [comment]
> where is the default inference configmap?

https://github.com/kaito-project/kaito/blob/main/charts/kaito/workspace/templates/inference-params.yaml  here.

File: pkg/workspace/inference/preset-inferences.go (hunk 1 above)
Comment (zhuangqh) [reply pkg/workspace/inference/preset-inferences.go:206]: sure

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**